src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -921,4 +922,60 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise. A date is valid if there is no dueDateError or assessmentDueDateError
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        // atleast one is set, so we have to check the two possible errors
+        boolean validDates = validateDueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate())
+                && validateAssessmentDueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        if (!validDates) {
+            throw new BadRequestException();
+        }
+    }
+
+    /**
+     * This method is used to validate the dueDate of an exercise. A due date is valid, if it is after the releaseDate,
+     * @return true if there is no DueDateError
+     */
+
+    private boolean validateDueDateError(ZonedDateTime releaseDate, ZonedDateTime dueDate, ZonedDateTime assesmentDueDate) {
```

Comment:
`assesmentDueDate` is not used in this method. Tbh I also find the method name a bit odd, since you're not validating the error but the date, so imho this should be `validateDueDate` (and `validateAsessmentDueDate` below).